### DISCUSSION TITLE
How can I translate this to nushell?

### DISCUSSION THREAD

**Discussion by andrewbanchich** [nushell/nushell#6435] [Q&A] [answered]
[This issue](https://github.com/purcell/exec-path-from-shell/issues/109#issuecomment-1229195853) has been preventing me from using `nushell` as my default shell, but I can't seem to figure out how to write the equivalent command.

The closest I've gotten is:

```
> echo $"sh -c /usr/bin/printf '__RESULT\u0000%s\u0000__RESULT' \"${MANPATH-08ccc6cf287832a08d51789038f15a72}\""
sh -c /usr/bin/printf '__RESULT%s__RESULT' "${MANPATH-08ccc6cf287832a08d51789038f15a72}"
```

But when I replace `echo` with `nu -c` I get:

```
❯ nu -c $"sh -c /usr/bin/printf '__RESULT\u0000%s\u0000__RESULT' \"${MANPATH-08ccc6cf287832a08d51789038f15a72}\""
Error: nu::shell::external_command (link)

  × External command failed
   ╭─[entry #37:1:1]
 1 │ nu -c $"sh -c /usr/bin/printf '__RESULT\u0000%s\u0000__RESULT' \"${MANPATH-08ccc6cf287832a08d51789038f15a72}\""
   · ─┬
   ·  ╰── can't run executable
   ╰────
  help: nul byte found in provided data
```

Per the linked issue:

> The goal is to print out __RESULT followed by a NULL character, followed by either the env var (if it exists) or the provided hex string (08ccc6cf287832a08d51789038f15a72 in this case), then finally a NULL and __RESULT again.

Any help would be appreciated. Thanks!

**Comment by fdncred** [comment]
You could construct something in nushell without using sh by doing something with nushell's `print` or `echo` command then using the `default` command an using `char -i 0` for the null character. If you do `help default` there's an example of how to do some of this with the `default` command.

**Reply by andrewbanchich** [reply]
That wouldn't work for this issue. The Emacs package that nushell breaks sets the PATH variable in Emacs to that of your shell, so it needs to run a shell command to get it. Since it can't support custom commands for every shell, the logic is "if it's bash or zsh, run the standard command - otherwise use sh -c".

**Reply by fdncred** [reply]
we just fixed a bug with quoted strings so you may want to build off the latest main branch to see if that fixes the problem.

**Reply by andrewbanchich** [reply]
Okay, I updated `nu` with `cargo install --git https://github.com/nushell/nushell --branch main --features extra nu` and got the same output as above.

**Reply by andrewbanchich** [reply]
> nul byte found in provided data

Any reason why nushell is explicitly failing for this?

**Reply by hustcer** [reply]
> > nul byte found in provided data
> 
> Any reason why nushell is explicitly failing for this?

You are running `nu -c $"sh -c /usr/bin/printf '__RESULT\u0000%s\u0000__RESULT' \"${MANPATH-08ccc6cf287832a08d51789038f15a72}\""` in nushell not in bash. In nu you can run `$"sh -c /usr/bin/printf '__RESULT\u0000%s\u0000__RESULT' \"${MANPATH-08ccc6cf287832a08d51789038f15a72}\""`

**Reply by hustcer** [reply]
`$"/usr/bin/printf '__RESULT\u0000%s\u0000__RESULT' \"${MANPATH-08ccc6cf287832a08d51789038f15a72}\"" | sh` Is this what you want?

**Reply by andrewbanchich** [reply]
Yep! That's perfect. Thank you @hustcer

**Reply by hustcer** [reply]
@andrewbanchich So is there anything else can prevent you from using nushell as your default shell ?

**Reply by andrewbanchich** [reply]
Not at the moment (after that Emacs package works again). It's been great!

**Comment by hustcer** [comment, marked as answer]
`$"/usr/bin/printf '__RESULT\u0000%s\u0000__RESULT' \"${MANPATH-08ccc6cf287832a08d51789038f15a72}\"" | sh` should work
